components/ModFacilitySubmissionLeftNavbar.vue

```diff
@@ -0,0 +1,161 @@
+                @click="scrollToSectionOfFacilityForm('change-log')"
+            >
+                {{ $t("modPanelFacilitySubmissionLeftNavbar.changeLog") }}
+            </button>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { ref, type Ref, onMounted, onUnmounted } from 'vue'
+import { useModerationSubmissionsStore } from '~/stores/moderationSubmissionsStore'
+
+const moderationSubmissionsStore = useModerationSubmissionsStore()
+const activeSection: Ref<string> = ref('contact-information')
+const isScrolling: Ref<boolean> = ref(false)
+
+const scrollToSectionOfFacilityForm = (sectionId: string) => {
+    document.getElementById(sectionId)?.scrollIntoView({
+        behavior: 'smooth'
+    })
+    activeSection.value = sectionId
+
+    isScrolling.value = true
+
+    setTimeout(() => {
+        isScrolling.value = false
+    }, 1000)
+}
+
+const observeFormSections = () => {
+    const sections: NodeListOf<Element> = document.querySelectorAll('.facility-form-section')
+
+    const addressObserver = new IntersectionObserver(
+        (entries: IntersectionObserverEntry[]) => {
+            let currentSection: string | null = null
+
+            entries.forEach((entry: IntersectionObserverEntry) => {
+                if (entry.isIntersecting && entry.intersectionRatio >= 0.5) {
+                    const sectionId = entry.target.id
+
+                    const threshold = sectionId === 'addresses' ? 0.5 : 1.0
+
+                    if (entry.intersectionRatio >= threshold && !isScrolling.value) {
+                        currentSection = sectionId
+                    }
+                }
+            })
+
+            if (currentSection) {
+                activeSection.value = currentSection
+            }
+        },
+        { threshold: [0.5, 1.0] }
+    )
+
+    const otherSectionsObserver = new IntersectionObserver(
+        (entries: IntersectionObserverEntry[]) => {
+            let currentSection: string | null = null
+
+            entries.forEach((entry: IntersectionObserverEntry) => {
+                if (entry.isIntersecting && entry.intersectionRatio >= 1.0) {
+                    const sectionId = entry.target.id
+                    if (!isScrolling.value) {
+                        currentSection = sectionId
+                    }
+                }
+            })
+
+            if (currentSection) {
+                activeSection.value = currentSection
+            }
+        },
+        { threshold: 1.0 }
+    )
+
+    sections.forEach(section => {
+        const sectionId = section.id
+        if (sectionId === 'addresses') {
+            addressObserver.observe(section)
+        } else {
+            otherSectionsObserver.observe(section)
+        }
+    })
+}
+
+const handleScroll = () => {
+    if (isScrolling.value) return
+
+    const sections: NodeListOf<Element> = document.querySelectorAll('.facility-form-section')
```

Comment:
you could make this reusable. Then instead of looking them up which has some risk of unwanted  entries, you could pass in an object array like:"
```
[{
    sectionTitle: "foo",
    sectionElementIdToScrollTo: 'bar'
 }, ...]
```

Reply:
reusable how? As in exported? I can do that or just locally? With this I would still need to get the element by ID. There are just so many ways I could do this I do not know which way you are looking for.